rcon/commands.py

```diff
@@ -581,18 +580,20 @@ def do_add_map_to_rotation(
         self,
         map_name: str,
         after_map_name: str = None,
-        after_map_name_number: str = None,
+        after_map_name_number: int = None,
     ):
-        cmd = f"rotadd {map_name}"
-        if after_map_name:
-            cmd = f"{cmd} {after_map_name}"
-            if after_map_name_number:
-                cmd = f"{cmd} {after_map_name_number}"
+        if after_map_name is None:
+            current = self.get_map_rotation()
+            after_map_name = current[len(current) - 1]
+            after_map_name_number = current.count(after_map_name)
+        cmd = f"rotadd /Game/Maps/{map_name} /Game/Maps/{after_map_name}"
+        if after_map_name_number:
+            cmd = f"{cmd} {after_map_name_number}"
```

Comment:
Only in extended_commands should `after_map_name` remain optional. `commands` is only a direct wrapper of the protocol's commands. And more importantly, you can not invalidate the response of `extended_commands`'s `get_map_rotation` like this.

Reply:
Good point 👍